frontend/app/src/App.test.tsx

```diff
@@ -219,1619 +369,1738 @@ describe("App", () => {
 
-    // @ts-expect-error
-    wrapper.instance().handleNewSession(new NewSession(newSessionJson))
-
-    expect(instance.clearAppState).toHaveBeenCalled()
-  })
-
-  it("doesn't call clearAppState if currentPageScriptHash doesn't change", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    wrapper.setState({
-      currentPageScriptHash: "page_script_hash",
-      appHash: hashString("installationId"),
+      // window.prerenderReady is set to true after first run
+      sendForwardMessage("sessionStatusChanged", {
+        runOnSave: false,
+        scriptIsRunning: true,
+      })
+      // @ts-expect-error
+      expect(window.prerenderReady).toBe(true)
     })
-    const instance = wrapper.instance() as App
-    instance.clearAppState = jest.fn()
 
-    const newSessionJson = cloneDeep(NEW_SESSION_JSON)
+    it("plumbs appPages and currentPageScriptHash to the AppView component", () => {
+      renderApp(getProps())
+      const hostCommunicationMgr = getStoredValue<HostCommunicationManager>(
+        HostCommunicationManager
+      )
 
-    // @ts-expect-error
-    wrapper.instance().handleNewSession(new NewSession(newSessionJson))
+      expect(screen.queryByTestId("stSidebarNav")).not.toBeInTheDocument()
 
-    expect(instance.clearAppState).not.toHaveBeenCalled()
-  })
+      const appPages = [
+        { pageScriptHash: "hash1", pageName: "page1" },
+        { pageScriptHash: "hash2", pageName: "page2" },
+      ]
 
-  it("sets hideSidebarNav based on new session message", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    const app = wrapper.instance() as App
+      sendForwardMessage("newSession", {
+        ...NEW_SESSION_JSON,
+        appPages,
+        pageScriptHash: "hash1",
+      })
 
-    // default in this.state is true
-    expect(app.state.hideSidebarNav).toBe(true)
+      expect(screen.getByTestId("stSidebarNav")).toBeInTheDocument()
+      const navLinks = screen.queryAllByTestId("stSidebarNavLink")
+      expect(navLinks).toHaveLength(2)
+      expect(navLinks[0]).toHaveStyle("font-weight: 600")
+      expect(navLinks[1]).toHaveStyle("font-weight: 400")
 
-    // update the value based on new session message
-    // @ts-expect-error
-    wrapper.instance().handleNewSession(new NewSession(NEW_SESSION))
-    expect(app.state.hideSidebarNav).toBe(false)
-  })
+      expect(document.title).toBe("page1 · Streamlit")
+      expect(hostCommunicationMgr.sendMessageToHost).toHaveBeenCalledWith({
+        type: "SET_APP_PAGES",
+        appPages,
+      })
+      expect(hostCommunicationMgr.sendMessageToHost).toHaveBeenCalledWith({
+        type: "SET_CURRENT_PAGE_NAME",
+        currentPageName: "",
+        currentPageScriptHash: "hash1",
+      })
+    })
 
-  describe("page change URL handling", () => {
-    let wrapper: ShallowWrapper
-    let instance: App
-    let pushStateSpy: any
+    it("calls clearAppState if currentPageScriptHash changes", async () => {
```

Comment:
Nit; Think we could revise the test name for clarity since its not directly testing the call to `clearAppState` function